libs/networkInterfaceImpl/src/main/java/com/solanamobile/mintyfresh/networkinterfaceimpl/usecase/MyMintsUseCase.kt

```diff
@@ -44,11 +45,15 @@ class MyMintsUseCase @Inject constructor(
                     clusterName = clusterName
                 )
                 if (cachedMint == null) {
-                    val metadata = nftRepository.getNftsMetadata(nft)
-                    val mint = metaplexToCacheMapper.map(nft, metadata, clusterName)
+                    try {
+                        val metadata = nftRepository.getNftsMetadata(nft)
+                        val mint = metaplexToCacheMapper.map(nft, metadata, clusterName)
 
-                    if (mint != null) {
-                        myMintsCacheRepository.insertAll(listOf(mint))
+                        if (mint != null) {
+                            myMintsCacheRepository.insertAll(listOf(mint))
+                        }
+                    } catch (e: Exception) {
+                        Log.e("MintyFresh", "Error loading NFT", e)
```

Comment:
@ankur2136 here is my justification for doing things this way:

- The most common errors are in the metaplex sdk, but if metaplex errors out, we don't want to put anything in the database anyway, so it's all the same operationally. 
- I think there are a couple possible errors that can result
- I don't think there's any useful viewstate to present to the user in this case. For the most part these errors are edge cases that only happen sporadically when the API times out or something of the sort. Usually a refresh will work just fine subsequent refreshes. 
- This is actually already way better functionality since before this error would propagate up to the my mints UI which would clear out the entire grid and display a generic errro.

Reply:
I am not saying we shouldn't do this. (We definitely should) 
I am saying maybe a better way would be to propagate the actual error instead of using `getOrThrow` in `NftRepository`

```
    suspend fun getNftsMetadata(nft: NFT) = withContext(Dispatchers.IO) {
        return@withContext nft.metadata(nftInfraFactory.storageDriver).getOrThrow()
    }
```

this is throwing unnecessarily and creating issues for consumers. It doesnt even tell the consumers of the API what actual error happened. Maybe use `getOrNull` instead.

Reply:
The two errors I've seen coming out of the metaplex SDK are:

1. Timeout
2. Weird collisions in Okhttp; likely will get fixed eventually

I mean we _could_ propagate some of that information up the consumer hierarchy, but again, the happy path is really what matters. If this NFT doesn't work, we still want the others to work. Just not seeing the benefit of granularity right now.